activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -916,6 +920,13 @@ def set_value(name, value) # :nodoc:
 
     private
 
+      # Extract column names from arguments passed to #order or #reorder.
+      def column_names_from_order_arguments(args)
+        args = args.dup
+        orderings = args.extract_options!
```

Comment:
The keys of the hash we're extracting are dependent on the schema (column names). I don't understand how this could be written to use kwargs.